File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var minBitwiseArray = function(nums) {
    let n=nums.length;
    let ans=[];
    for(let i=0;i<n;i++){
        let flag=false;
        let x=nums[i];
        for(let j=0;j<x;j++){
            if((j | (j+1))==x){
                flag=true;
                ans.push(j);
                break;
            }
        }
        if(!flag)ans.push(-1);
    }
    return ans;
};

File: 3605-construct-the-minimum-bitwise-array-i/construct-the-minimum-bitwise-array-i.cpp
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
                int n=nums.size();

        vector<int>ans(n,INT_MAX);
        for(int i=0;i<n;i++){
            int x=nums[i];
            int rem=0;
            bool flag=false;
            for(int j=1;j<x;j++){
                if((j | (j+1))==x){
                    ans[i]=(j);
                    flag=true;
                    break;
                }
            }
            if(!flag)ans[i]=-1;
            
        }
        return ans;
    }
};